### ISSUE TITLE
[turborepo] SIGSEGV error while accessing remote cache

### ISSUE THREAD

**Issue by kheyse-oqton** [vercel/turbo#5055]
### What version of Turborepo are you using?

1.9.6

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Linux

### Describe the Bug

Crash with SIGSEGV during yarn build while building dockerimage with TURBO_API, TURBO_TEAM and TURBO_TOKEN configured. Without these env vars, there's no SIGSEGV.

```
#12 [builder 6/7] RUN yarn turbo build --filter=@*****/abc
#12 sha256:960b24b7a1bfc0419e7be0354e2bdc6863effe5b8fecddadd751959bd904bff6
#12 0.506 yarn run v1.22.19
#12 0.531 $ /usr/src/app/node_modules/.bin/turbo build --filter=@*****/abc
#12 1.818 • Packages in scope: @*****/abc
#12 1.818 • Running build in 1 packages
#12 1.818 • Remote caching enabled
#12 1.822 panic: runtime error: invalid memory address or nil pointer dereference
#12 1.822 [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xca4daa]
#12 1.822 
#12 1.822 goroutine 16666 [running]:
#12 1.822 github.com/vercel/turbo/cli/internal/client.(*APIClient).getArtifact(0xc000230000, {0xc000c760e0, 0x10}, {0x350f09, 0x3})
#12 1.823 	github.com/vercel/turbo/cli/internal/client/cache.go:114 +0x5ca
#12 1.823 github.com/vercel/turbo/cli/internal/client.(*APIClient).FetchArtifact(0x40?, {0xc000c760e0?, 0xc0009048e0?})
#12 1.823 	github.com/vercel/turbo/cli/internal/client/cache.go:75 +0x2a
#12 1.823 github.com/vercel/turbo/cli/internal/cache.(*httpCache).retrieve(0xc0002280a0, {0xc000c760e0, 0x10})
#12 1.823 	github.com/vercel/turbo/cli/internal/cache/cache_http.go:202 +0x7e
#12 1.823 github.com/vercel/turbo/cli/internal/cache.(*httpCache).Fetch(0xc0002280a0, {0xc00022a0d0?, 0xc?}, {0xc000c760e0, 0x10}, {0x1?, 0xc000904ba8?, 0x8ee0c5?})
#12 1.823 	github.com/vercel/turbo/cli/internal/cache/cache_http.go:150 +0xdb
#12 1.823 github.com/vercel/turbo/cli/internal/cache.(*cacheMultiplexer).Fetch(0xc000144070, {0xc00022a0d0, 0xc}, {0xc000c760e0, 0x10}, {0x0, 0x0, 0x0})
#12 1.823 	github.com/vercel/turbo/cli/internal/cache/cache.go:258 +0x1a9
#12 1.823 github.com/vercel/turbo/cli/internal/cache.(*asyncCache).Fetch(0xc0003c6570?, {0xc00022a0d0?, 0x0?}, {0xc000c760e0?, 0x0?}, {0x0?, 0xc0003c6570?, 0x17?})
#12 1.823 	github.com/vercel/turbo/cli/internal/cache/async_cache.go:54 +0x31
#12 1.823 github.com/vercel/turbo/cli/internal/runcache.(*TaskCache).RestoreOutputs(0xc00054b388, {0x4642c0, 0xc000036038}, 0xc000228190?, {0x468138?, 0xc000142510?})
#12 1.823 	github.com/vercel/turbo/cli/internal/runcache/runcache.go:135 +0x396
#12 1.823 github.com/vercel/turbo/cli/internal/run.(*execContext).exec(0xc0002918c0, {0x4642c0, 0xc000036038}, 0xc0000f6f00)
#12 1.823 	github.com/vercel/turbo/cli/internal/run/real_run.go:263 +0x536
#12 1.823 github.com/vercel/turbo/cli/internal/run.RealRun.func2({0x4642c0?, 0xc000036038?}, 0xc0003c6330?, 0xc00033c400)
#12 1.823 	github.com/vercel/turbo/cli/internal/run/real_run.go:101 +0x77
#12 1.823 github.com/vercel/turbo/cli/internal/graph.(*CompleteGraph).GetPackageTaskVisitor.func1({0xc0003c6330, 0x15})
#12 1.823 	github.com/vercel/turbo/cli/internal/graph/graph.go:174 +0xfbc
#12 1.823 github.com/vercel/turbo/cli/internal/core.(*Engine).Execute.func2.1(0xc000d08ff0, {0x25c1e0?, 0xc0006c3700?}, {0x60?, 0x8e7a45?}, 0xc0003d39e0, 0xc000bc3c20, 0xc0008ac2d0)
#12 1.823 	github.com/vercel/turbo/cli/internal/core/engine.go:138 +0xef
#12 1.823 github.com/vercel/turbo/cli/internal/core.(*Engine).Execute.func2({0x25c1e0?, 0xc0006c3700?})
#12 1.823 	github.com/vercel/turbo/cli/internal/core/engine.go:151 +0x45
#12 1.823 github.com/pyr-sh/dag.(*Walker).walkVertex(0xc000bc3ce0, {0x25c1e0, 0xc0006c3700}, 0xc00031a300)
#12 1.823 	github.com/pyr-sh/dag@v1.0.0/walk.go:377 +0x2fa
#12 1.823 created by github.com/pyr-sh/dag.(*Walker).Update
#12 1.823 	github.com/pyr-sh/dag@v1.0.0/walk.go:300 +0xf45
```

### Expected Behavior

Don't crash while accessing remote cache.

### To Reproduce

Hope this dockerfile excerpt is sufficient to reproduce.
```
ARG NODE_IMAGE=node:18.13-bullseye-slim

FROM ${NODE_IMAGE} 

WORKDIR /usr/src/app

COPY . .

ENV CI=true

RUN yarn install --frozen-lockfile

ARG TURBO_API
ENV TURBO_API=$TURBO_API

ARG TURBO_TEAM
ENV TURBO_TEAM=$TURBO_TEAM

ARG TURBO_TOKEN
ENV TURBO_TOKEN=$TURBO_TOKEN

RUN yarn turbo build --filter=@oqton/abc
```

<sub>[TURBO-1492](https://linear.app/vercel/issue/TURBO-1492/[turborepo]-sigsegv-error-while-accessing-remote-cache)</sub>

**Comment by NicholasLYang** [comment]
Hi @kheyse-oqton, thanks for the issue. Could you try with a different TURBO_TOKEN and see if it still errors? I have a hunch as to what's causing the segfault.

**Comment by kheyse-oqton** [comment]
> Hi @kheyse-oqton, thanks for the issue. Could you try with a different TURBO_TOKEN and see if it still errors? I have a hunch as to what's causing the segfault.

Using a different, invalid TURBO_TOKEN produces the same error.

**Comment by kheyse-oqton** [comment]
It appears that just the existence of our TURBO_API env value causes the issue, even if token and team are not set.

**Comment by kheyse-oqton** [comment]
The SIGSEGV appears to be triggered when TURBO_API starts with an (accidental) `"`. Has nothing to do with docker or token. Hope this is sufficient to reproduce and fix.

**Comment by NicholasLYang** [comment]
Gotcha. Yeah my hunch was that it was some funkiness with Go's net/http module, since there have been reports of [segfaults with HTTP headers including newlines](https://www.jvt.me/posts/2022/02/28/go-segfault-http-newline/)

**Comment by arlyon** [comment]
We have released 1.11 with rust by default so the codepath that causes the segfault can no longer happen. Please comment if you notice this issue on 1.11
